contrib/action_recognition/r2p1d/vu/models/r2plus1d.py

```diff
@@ -0,0 +1,401 @@
+            temporal_jitter=True if cfgs.get('temporal_jitter_step', 0) > 0 else False,
+            flip_ratio=cfgs.get('flip_ratio', 0.5),
+            random_crop=cfgs.get('random_crop', True),
+            random_crop_scales=cfgs.get('random_crop_scales', (0.6, 1.0)),
+            video_ext=cfgs.video_ext,
+        )
+
+        valid_split = cfgs.get('valid_split', None)
+        valid_ds = None if valid_split is None else VideoDataset(
+            split_file=valid_split,
+            video_dir=cfgs.video_dir,
+            num_segments=1,
+            sample_length=cfgs.sample_length,
+            sample_step=cfgs.get('sample_step', 1),
+            input_size=112,
+            im_scale=cfgs.get('im_scale', 128),
+            resize_keep_ratio=True,
+            mean=cfgs.get('mean', DEFAULT_MEAN),
+            std=cfgs.get('std', DEFAULT_STD),
+            random_shift=False,
+            temporal_jitter=False,
+            flip_ratio=0.0,
+            random_crop=False,  # == Center crop
+            random_crop_scales=None,
+            video_ext=cfgs.video_ext,
+        )
+
+        return train_ds, valid_ds
+
+    def show_batch(self, which_data='train', num_samples=1):
+        """Plot first few samples in the datasets"""
+        if which_data == 'train':
+            batch = [self.train_ds[i][0] for i in range(num_samples)]
+        elif which_data == 'valid':
+            batch = [self.valid_ds[i][0] for i in range(num_samples)]
+        else:
+            raise ValueError("Unknown data type {}".format(which_data))
+        _show_batch(
+            batch,
+            self.configs.sample_length,
+            mean=self.configs.get('mean', DEFAULT_MEAN),
+            std=self.configs.get('std', DEFAULT_STD),
+        )
+
+    def freeze(self):
+        """Freeze model except the last layer"""
+        self._set_requires_grad(False)
+        for param in self.model.fc.parameters():
+            param.requires_grad = True
+
+    def unfreeze(self):
+        self._set_requires_grad(True)
+
+    def _set_requires_grad(self, requires_grad=True):
+        for param in self.model.parameters():
+            param.requires_grad = requires_grad
+
+    def fit(self, train_cfgs):
+        train_cfgs = Config(train_cfgs)
+
+        model_dir = train_cfgs.get('model_dir', "checkpoints")
+        os.makedirs(model_dir, exist_ok=True)
+
+        if cuda.is_available():
+            device = torch.device("cuda")
+            num_gpus = cuda.device_count()
+            # Look for the optimal set of algorithms to use in cudnn. Use this only with fixed-size inputs.
+            torch.backends.cudnn.benchmark = True
+        else:
+            device = torch.device("cpu")
+            num_gpus = 0
+
+        data_loaders = {}
+        if self.train_ds is not None:
+            data_loaders['train'] = DataLoader(
+                self.train_ds,
+                batch_size=train_cfgs.get('batch_size', 8) * num_gpus,
```

Comment:
If no GPUs are available, num_gpus will be 0 (from line 175) and the batch_size will also be 0

Reply:
Nice catch! I will fix this soon.